Blackjax currently broken in py3.10

### Describe the issue as clearly as possible:

Running test in a fresh python 3.10 environment returns:

### Steps/code to reproduce the bug:

```python
n/a
```


### Expected result:

```shell
n/a
```


### Error message:

```shell
ImportError while importing test module '/Users/junpenglao/Documents/OSS/blackjax/tests/test_uturn.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/miniconda3/envs/blackjax/lib/python3.10/site-packages/_pytest/python.py:618: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/opt/miniconda3/envs/blackjax/lib/python3.10/site-packages/_pytest/pathlib.py:533: in import_path
    importlib.import_module(module_name)
/opt/miniconda3/envs/blackjax/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/opt/miniconda3/envs/blackjax/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
tests/test_uturn.py:6: in <module>
    from blackjax.mcmc.metrics import gaussian_euclidean
blackjax/__init__.py:3: in <module>
    from .kernels import (
blackjax/kernels.py:20: in <module>
    import blackjax.adaptation as adaptation
blackjax/adaptation/__init__.py:1: in <module>
    from . import meads, pathfinder_adaptation, window_adaptation
blackjax/adaptation/pathfinder_adaptation.py:19: in <module>
    from blackjax.adaptation.step_size import (
blackjax/adaptation/step_size.py:20: in <module>
    import blackjax.optimizers.dual_averaging as dual_averaging
E   ImportError: cannot import name 'optimizers' from partially initialized module 'blackjax' (most likely due to a circular import) (/Users/junpenglao/Documents/OSS/blackjax/blackjax/__init__.py)
```


### Blackjax/JAX/jaxlib/Python version information:

```python
blackjax main at HEAD
Python 3.10
```


### Context for the issue:

_No response_

--- a/blackjax/adaptation/step_size.py
+++ b/blackjax/adaptation/step_size.py
@@ -17,7 +17,7 @@ from typing import Callable, NamedTuple, Tuple
 import jax
 import jax.numpy as jnp
 
-import blackjax.optimizers.dual_averaging as dual_averaging
+from blackjax.optimizers.dual_averaging import dual_averaging
 from blackjax.mcmc.hmc import HMCState
 
 __all__ = [
@@ -122,7 +122,7 @@ def dual_averaging_adaptation(
            of Machine Learning Research 15.1 (2014): 1593-1623.
 
     """
-    da_init, da_update, da_final = dual_averaging.dual_averaging(t0, gamma, kappa)
+    da_init, da_update, da_final = dual_averaging(t0, gamma, kappa)
 
     def init(inital_step_size: float) -> DualAveragingAdaptationState:
         """Initialize the state of the dual averaging scheme.

--- a/blackjax/adaptation/step_size.py
+++ b/blackjax/adaptation/step_size.py
@@ -17,8 +17,8 @@ from typing import Callable, NamedTuple, Tuple
 import jax
 import jax.numpy as jnp
 
-from blackjax.optimizers.dual_averaging import dual_averaging
 from blackjax.mcmc.hmc import HMCState
+from blackjax.optimizers.dual_averaging import dual_averaging
 
 __all__ = [
     "dual_averaging_adaptation",

--- a/blackjax/optimizers/dual_averaging.py
+++ b/blackjax/optimizers/dual_averaging.py
@@ -20,6 +20,7 @@ __all__ = [
     "dual_averaging",
 ]
 
+
 class DualAveragingState(NamedTuple):
     """State carried through the dual averaging procedure.
 

--- a/blackjax/optimizers/dual_averaging.py
+++ b/blackjax/optimizers/dual_averaging.py
@@ -15,6 +15,10 @@ from typing import Callable, NamedTuple, Tuple
 
 import jax.numpy as jnp
 
+__all__ = [
+    "DualAveragingState",
+    "dual_averaging",
+]
 
 class DualAveragingState(NamedTuple):
     """State carried through the dual averaging procedure.

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,5 @@
 -e ./
 chex
-fastprogress>=0.2.0
 pre-commit
 pytest
 pytest-benchmark

--- a/setup.py
+++ b/setup.py
@@ -46,6 +46,7 @@ setuptools.setup(
         "jax>=0.3.13",
         "jaxlib>=0.3.10",
         "jaxopt>=0.5.5",
+        "typing-extensions>=4.4.0",
     ],
     long_description_content_type="text/markdown",
     keywords="probabilistic machine learning bayesian statistics sampling algorithms",

--- a/tests/test_optimizers.py
+++ b/tests/test_optimizers.py
@@ -10,7 +10,7 @@ from absl.testing import absltest, parameterized
 from jax.flatten_util import ravel_pytree
 from jaxopt._src.lbfgs import compute_gamma, inv_hessian_product
 
-import blackjax.optimizers.dual_averaging as dual_averaging
+from blackjax.optimizers.dual_averaging import dual_averaging
 from blackjax.optimizers.lbfgs import (
     lbfgs_inverse_hessian_factors,
     lbfgs_inverse_hessian_formula_1,
@@ -38,7 +38,7 @@ class OptimizerTest(chex.TestCase):
 
         # Our target gradient is 0. we increase the rate of convergence by
         # increasing the value of gamma (see documentation of the algorithm).
-        init, update, final = dual_averaging.dual_averaging(gamma=0.3)
+        init, update, final = dual_averaging(gamma=0.3)
         unpdate_fn = self.variant(update)
 
         da_state = init(3)

